### PR TITLE
Printable 10x11mm chain and magnetic probe for VSW

### PR DESCRIPTION
**Pull request by hymness1**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?
Quickdraw Probe remix for Voron Switchwire, with macros. There's also a printable and openable 10x11mm chain
#### How was it tested? 
I did, and some others users have tried the Quickdraw Probe.
#### Any background context you want to provide?
No
#### Screenshots (if appropriate)
There are in the folders
#### Further notes
Keep on the good work!

### CODE REVIEW

**Review by KiloQubit — COMMENTED** [review]
Thanks for your mods!  I see that you have a LICENSE file pointing to a different license.  All mods in this repository MUST be GPL3, and there's a license file at the top of the repository to that effect.  We cannot accept any mods here that are not GPL3, so please remove all non-GPL3 content as well as the license file.

**Comment by hymness1** [comment]
Hi KiloQubit,

I don't know much about GitHub, but I think I did what you asked me. All
files in my repo that I want merged in VoronUsers are my own and I created
them myself.

If you need anything else please let me know, as I said, I'm a GitNoob ;)
Thanks!

Le sam. 2 oct. 2021, à 13 h 12, KiloQubit ***@***.***> a
écrit :

**Comment by hymness1** [comment]
Hi, thanks at least for reaching out :) I'm trying to do things the right
way. I'll reach out at Annex and see if everything would be fine and come
back at you after if that's okay with you.

The chains however are my own, so there shouldn't be any problem. But I
don't know if I'd have to manually do a pull request for that.

Thanks so much for your patience, I suppose I'm not the first noob you
encounter ;)

Le mar. 5 oct. 2021, à 21 h 39, KiloQubit ***@***.***> a
écrit :

File: printer_mods/README.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ like so:
 | |[Din Rail mount](./hijax_pl/din_rail_mount) |Adapter to attach DIN Rail to 2020 extrusion. |:x: |:x: |:heavy_check_mark: |:x:|
 | |[Controler Board adapters](./hijax_pl/ender_compatible_controller_mount) |DIN (rotated) and legacy adapters for boards like SKR mini E3. |:grey_question: |:x: |:heavy_check_mark: |:x:|
 |hoschi| [FYSETC Spider DIN Rail Mount](./hoschi/FYSETC_Spider_DIN_Mount) | DIN rail mount for FYSETC Spider | :x: | :grey_question: | :heavy_check_mark: | :grey_question: |
+|hymness1| [Quickdraw_probe_Voron_Switchwire](./hymness1/Quickdraw_probe_Voron_Switchwire) | Detachable magnetic probe with docking system for VSW | :x: | :x: | :x: | :heavy_check_mark: |
```
Comment (KiloQubit) [inline printer_mods/README.md:137]: Please add an `:x:` in the Trident column.

Please remove underscores from inside square brackets, that is the text shown to the user:

    [Quickdraw probe Voron Switchwire](./hymness1/Quickdraw_probe_Voron_Switchwire)

Same thing on next line.

File: printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+You first need to enter the parameters for the dock variables in dock_macros.cfg. The values that are in the config file are the values I use for my Switchwire. You can still use these macros if you use the files for the probe and dock from Annex Engineering. If you do, you'll want to change the value for variable_detach from 1 to 2.
+
+For the initial homing for the Z-axis, there are options for probe homing, sensorless homing and endstop microswitch homing.
+
+You need to add [include dock_macros.cfg] in your printer.cfg file
+
+There are some safety measures but this is still a work in progress. Be sure to add dock_probe to your start_print macro.
+
+A tutorial to get all this working will be available [here](https://docs.google.com/document/d/1dSgqjHQ3RNvcPwyRcuyMOceWPvVQUWRSjj3-KcRRMoo/edit?usp=sharing)
```
Comment (KiloQubit) [inline printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md:11]: This links to an empty document for me.

File: printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+The Switchwire has about 230mm of print area in the Y direction, but has around 250mm of Y travel (mine has 257mm), from the endstop to the endblock. About 20 of those millimeters are located outside the bed at the front when the bed is pushed all the way to the rear of the printer. At this location, the nozzle and the part cooling ducts can be clear of the bed when homing at Z_min. The default `printer.cfg` for the VSW with an SKR Mini E3 V2 calls for 0 as position_min and 250 as position_max, whilst it should be, at least that is my case, -25 as position_min and 232 as position_max. When populating the variables in `dock_macros.cfg`, be sure to enter an existing value for `variable_y_min`, and that this value is located outside the bed. If you are using default config for `printer.cfg`, 0 should be a safe coordinate. If you changed the config file for the real travel in the Y axis, -20 should be a safe coordinate. Failure to enter an existing or safe value could at best prevent you from homing with the message `Move out of range`, or at worst crash the nozzle into the bed.
+
+![SW_Y_travel](/printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Pictures/SW_Y_travel.png)
```
Comment (KiloQubit) [inline printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md:76]: Can you make this a relative reference from this directory?  i.e. it should be

    SW_Y_travel

**Review by KiloQubit — CHANGES_REQUESTED** [review]
Thanks for your updates!

Looks like there's another unsupported overhang that needs either a chamfer or a built-in support in `chainlink.stl`
Screen Shot 2021-10-16 at 14 06 40

Too steep overhang in `fixed_end.stl`
Screen Shot 2021-10-16 at 14 08 40

**Comment by hymness1** [comment]
Thanks for your feedback. Modifications have been done :)

Le sam. 16 oct. 2021, à 14 h 27, KiloQubit ***@***.***> a
écrit :

File: printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md (hunk 3 above)
Comment (KiloQubit) [reply printer_mods/hymness1/Quickdraw_probe_Voron_Switchwire/Klipper_Macros/README.md:76]: Thanks.  Can you also update this README image link?

**Comment by hymness1** [comment]
Done

Le sam. 16 oct. 2021, à 16 h 23, KiloQubit ***@***.***> a
écrit :

**Review by KiloQubit — APPROVED** [review]
Thanks for the updates!

**Pull request merged**